### PR TITLE
Introduce blobstore copy client

### PR DESCRIPTION
**Pull request by xbrianh**
depends on #299 
related to #294

### CODE REVIEW

File: terra_notebook_utils/blobstore/copy_client.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+from getm.concurrent.collections import ConcurrentHeap
+
+from terra_notebook_utils.blobstore.local import LocalBlobStore, LocalBlob
+from terra_notebook_utils.blobstore.gs import GSBlobStore, GSBlob
+from terra_notebook_utils.blobstore import BlobstoreChecksumError
+
+
+logger = logging.getLogger(__name__)
+
+AnyBlobStore = Union[LocalBlobStore, GSBlobStore]
```
Comment (DailyDreaming) [inline terra_notebook_utils/blobstore/copy_client.py:17]: Guessing the URLBlobstore addition will be a separate PR?

**Review by DailyDreaming — APPROVED** [review]
LGTM.

File: terra_notebook_utils/blobstore/copy_client.py (hunk 1 above)
Comment (xbrianh) [reply terra_notebook_utils/blobstore/copy_client.py:17]: ugh will add now

**Review by DailyDreaming — APPROVED** [review]
